packages/seacas/libraries/ioss/src/Ioss_Decomposition.h

```diff
@@ -599,59 +616,60 @@ namespace Ioss {
                          m_comm);
       show_progress("\tCommunication 1 finished");
 
-      std::vector<size_t> ioss_offset =
-          Ioss::get_all_block_ioss_offset(blocks, block_component_count);
-      std::vector<size_t> import_offset =
-          Ioss::get_all_block_import_offset(blocks, block_component_count);
+      Ioss::ElementBlockBatchOffset batchOffset(blocks);
+      std::vector<size_t> iossOffset = batchOffset.get_ioss_offset(blockSubsetIndex, blockComponentCount);
+      std::vector<size_t> importOffset = batchOffset.get_import_offset(blockSubsetIndex, blockComponentCount);
 
       // Map local and imported data to ioss_data.
-      for (size_t blk_seq = 0; blk_seq < blocks.size(); blk_seq++) {
-        const Ioss::BlockDecompositionData &block      = blocks[blk_seq];
-        size_t                              comp_count = block_component_count[blk_seq];
+      for(size_t bsi = 0; bsi < blockSubsetIndex.size(); bsi++) {
+        int64_t blk_seq = blockSubsetIndex[bsi];
+        const Ioss::BlockDecompositionData& block = blocks[blk_seq];
+        size_t comp_count = blockComponentCount[bsi];
 
         for (size_t i = 0; i < block.localMap.size(); i++) {
           for (size_t j = 0; j < comp_count; j++) {
-            unsigned fileIndex = file_offset[blk_seq] + block.localMap[i] * comp_count + j;
-            unsigned iossIndex =
-                ioss_offset[blk_seq] + (i + block.localIossOffset) * comp_count + j;
+            unsigned fileIndex = fileOffset[bsi] + block.localMap[i] * comp_count + j;
```

Comment:
Is `unsigned` going to be large enough for big models or should `size_t` be used?

Reply:
You're right ... those are constructed from data that is `size_t` so they should be declared as `size_t`